### PR TITLE
Storybook

### PR DESCRIPTION
**Pull request by ptbrowne**
## Goals

- Extract component documentation out of catalog
- Catalog is only kept for "application user guide" right now. It could be completely moved to storybook in a second step.

## Comments

- Tried to use vite but it did not work well
- Used where applicable .mock.ts files that are used inside stories. In the future, we should take a look at how we could
  share more mocks between components
- Storybook can be sometimes a bit unstable and hang, see "Common errors" section
- Could reuse the "theme.stories.tsx" from "uis" :+1:
- I tried to follow the components / organisms / pages nomenclature
- Storybook is built separate from the main application. It is built into the public/ folder so that it is available via /storybook.

## Common errors when developing on Storybook

- Providers do not work
  - Need to wrap to use Story or Canvas so that they are wrapped by the decorators configured in preview.ts
- Canvas' `of` does not work
  - Need to use <Meta /> in the docs file
  - Story need to be loaded in main.ts (beware of tsx/mdx extension). This error was because during developement, I manually added one by one each story, otherwise there were errors. Now that there is a glob for all *.docs.mdx and *.stories.tsx, it should be OK.

- At least one component must be exported otherwise the story file can be reported missing ()
- Sometimes, loading a story hangs. Restarting storybook, and sometimes the browser helps.

### CODE REVIEW

File: app/docs/chart-selection-tabs.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      {
+        ...palmerPenguinsFixture.data,
+        state: "PUBLISHED",
+      }
+    }
+    allowDefaultRedirect={false}
+  >
+    <ChartSelectionTabs />
+  </ConfiguratorStateProvider>
+</ReactSpecimen> */}
```
Comment (bprusinowski) [inline app/docs/chart-selection-tabs.docs.mdx]: 👀

File: app/docs/fixtures.ts [hunk 2]
```diff
@@ -475,55 +475,91 @@ export const tableObservations = [
   },
 ];
 
-export const tableMeasures = [
+export const tableMeasures: Measure[] = [
```
Comment (bprusinowski) [inline app/docs/fixtures.ts:478]: 👍

**Review by bprusinowski — APPROVED** [review]
Super cool, thanks! 🥇 I'll also take a look at the deployment preview once it boots up 👍

**Comment by adintegra** [comment]
Very cool @ptbrowne 🎉 Thanks for this!

**Pull request merged**